dvc/remote/base.py

```diff
@@ -925,15 +946,19 @@ def checkout(
         if failed or skip:
             if progress_callback:
                 progress_callback(
-                    str(path_info), self.get_files_number(checksum)
+                    str(path_info),
+                    self.get_files_number(
+                        self.path_info, checksum, filter_info
+                    ),
```

Comment:
get_files_number logic is weird in general. We could show the progress bar for the outputs and for files inside of the directory. That would be much more straightforward logically.